include/libopencm3/nrf5/common/clock_common.h

```diff
@@ -0,0 +1,224 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* THIS FILE SHOULD NOT BE INCLUDED DIRECTLY, BUT ONLY VIA CLOCK.H
+The order of header inclusion is important. clock.h includes the device
+specific memorymap.h header before including this header file.
+*/
+
+/** @cond */
+#ifdef LIBOPENCM3_CLOCK_H
+/** @endcond */
+
+#ifndef LIBOPENCM3_CLOCK_COMMON_H
+#define LIBOPENCM3_CLOCK_COMMON_H
+
+/**@{*/
+
+/** @defgroup clock_registers Clock Control Registers
+ * @ingroup clock_defines
+ * @brief Clock Control Registers
+@{*/
+
+/* --------------------- CLOCK registers for NRF52 ------------------------- */
+/** Tasks Registers */
+#define CLOCK_TASKS_HFCLKSTART                  MMIO32(CLOCK_BASE + 0x000)
+#define CLOCK_TASKS_HFCLKSTOP                   MMIO32(CLOCK_BASE + 0x004)
+#define CLOCK_TASKS_LFCLKSTART                  MMIO32(CLOCK_BASE + 0x008)
+#define CLOCK_TASKS_LFCLKSTOP                   MMIO32(CLOCK_BASE + 0x00C)
+#define CLOCK_TASKS_CAL                         MMIO32(CLOCK_BASE + 0x010)
+#define CLOCK_TASKS_CTSTART                     MMIO32(CLOCK_BASE + 0x014)
+#define CLOCK_TASKS_CTSTOP                      MMIO32(CLOCK_BASE + 0x018)
+
+/** Events Registers */
+#define CLOCK_EVENTS_HFCLKSTARTED               MMIO32(CLOCK_BASE + 0x100)
+#define CLOCK_EVENTS_LFCLKSTARTED               MMIO32(CLOCK_BASE + 0x104)
+#define CLOCK_EVENTS_DONE                       MMIO32(CLOCK_BASE + 0x10C)
+#define CLOCK_EVENTS_CTTO                       MMIO32(CLOCK_BASE + 0x110)
+
+/** Registers */
+#define CLOCK_INTENSET                          MMIO32(CLOCK_BASE + 0x304)
+#define CLOCK_INTENCLR                          MMIO32(CLOCK_BASE + 0x308)
+#define CLOCK_HFCLKRUN                          MMIO32(CLOCK_BASE + 0x408)
+#define CLOCK_HFCLKSTAT                         MMIO32(CLOCK_BASE + 0x40C)
+#define CLOCK_LFCLKRUN                          MMIO32(CLOCK_BASE + 0x414)
+#define CLOCK_LFCLKSTAT                         MMIO32(CLOCK_BASE + 0x418)
+#define CLOCK_LFCLKSRCCOPY                      MMIO32(CLOCK_BASE + 0x41C)
+#define CLOCK_LFCLKSRC                          MMIO32(CLOCK_BASE + 0x518)
+#define CLOCK_CTIV                              MMIO32(CLOCK_BASE + 0x538)
+#define CLOCK_TRACECONFIG                       MMIO32(CLOCK_BASE + 0x55C)
+/**@}*/
+
+/* --- CLOCK_INTENSET values ----------------------------------------------- */
+/** @defgroup clock_intenset Enable interrupt
+@ingroup clock_defines
+@{*/
+#define CLOCK_INTENSET_HFCLKSTARTED                (1 << 0)
+#define CLOCK_INTENSET_LFCLKSTARTED                (1 << 1)
+#define CLOCK_INTENSET_DONE                        (1 << 3)
+#define CLOCK_INTENSET_CTTO                        (1 << 4)
+/**@}*/
+
+/* --- CLOCK_INTENCLR values ----------------------------------------------- */
+/** @defgroup clock_intenset Disable interrupt
```

Comment:
duplicate group here.

Reply:
(iirc, you don't need the ingroup, that's automatic because of where you are in the file)